torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -52,8 +59,15 @@ void Polynomial::sort() {
   if (dtype().is_floating_point()) {
     throw std::logic_error("reordering FP ops");
   }
+  std::unordered_map<ExprPtr, std::string> str_repr_cache;
   std::sort(variables_.begin(), variables_.end(), [&](ExprPtr a, ExprPtr b) {
-    return hasher_.hash(a) < hasher_.hash(b);
+    if (!str_repr_cache.count(a)) {
```

Comment:
This seems to be useful enough to have it as a method. Just a suggestion. I'll leave it to you though.